### PR TITLE
Issue #3062474 by wengerk: Ensure EventTrackerService get the Price using any available Price Resolver

### PR DESCRIPTION
**Pull request by WengerK**
https://www.drupal.org/project/commerce_google_tag_manager/issues/3062474

### CODE REVIEW

File: src/EventTrackerService.php [hunk 1]
```diff
@@ -292,6 +328,9 @@ private function buildProductFromOrderItem(OrderItemInterface $order_item) {
    *   The Enhanced Ecommerce product.
    */
   private function buildProductFromProductVariation(ProductVariationInterface $product_variation) {
+    $context = new Context($this->currentUser, $this->currentStore->getStore());
+    /** @var \Drupal\commerce_price\Price $price */
```
Comment (WengerK) [inline src/EventTrackerService.php]:
```suggestion
```

File: src/EventTrackerService.php [hunk 2]
```diff
@@ -301,7 +340,9 @@ private function buildProductFromProductVariation(ProductVariationInterface $pro
     // Ensure the price is available before using it.
     $price = $product_variation->getPrice();
```
Comment (WengerK) [inline src/EventTrackerService.php]:
```suggestion
    /** @var \Drupal\commerce_price\Price $price */
    $price = $product_variation->getPrice();
```

File: src/EventTrackerService.php [hunk 3]
```diff
@@ -301,7 +340,9 @@ private function buildProductFromProductVariation(ProductVariationInterface $pro
...
     if ($price) {
-      $product->setPrice($this->formatPrice((float) $price->getNumber()));
+      // Ensure the price is available before using it.
+      $calculated_price = $this->priceCalculator->calculate($product_variation, 1, $context)->getCalculatedPrice();
```
Comment (WengerK) [inline src/EventTrackerService.php]:
```suggestion
	  /** @var \Drupal\commerce_price\Price $calculated_price */
      $calculated_price = $this->priceCalculator->calculate($product_variation, 1, $context)->getCalculatedPrice();
```

File: src/EventTrackerService.php [hunk 4]
```diff
@@ -292,16 +328,22 @@ private function buildProductFromOrderItem(OrderItemInterface $order_item) {
...
       ->setName($product_variation->getProduct()->getTitle())
       ->setId($product_variation->getProduct()->id())
       ->setVariant($product_variation->getTitle());
 
     // Ensure the price is available before using it.
+    /** @var \Drupal\commerce_price\Price $price */
     $price = $product_variation->getPrice();
     if ($price) {
-      $product->setPrice($this->formatPrice((float) $price->getNumber()));
+      // Ensure the price is available before using it.
```
Comment (WengerK) [inline src/EventTrackerService.php]:
```suggestion
```

File: tests/src/Kernel/EventTrackerServiceTest.php [hunk 5]
```diff
@@ -72,6 +73,30 @@ public function testBuildProductFromProductVariation() {
...
+   */
+  public function testBuildProductFromProductVariationCalculatedPrice() {
+    // The variations to test with.
+    $variation = ProductVariation::create([
+      'type' => 'default',
+      // Only product starting with TEST_ will pass
+      // the Drupal\commerce_price_test\TestPriceResolver.
+      'sku' => 'TEST_CALCULATED_PRICE2',
+      'status' => TRUE,
+      'price'  => new Price('11.00', 'USD'),
```
Comment (WengerK) [inline tests/src/Kernel/EventTrackerServiceTest.php]:
```suggestion
      'price' => new Price('11.00', 'USD'),
```

**Review by WengerK — COMMENTED** [review]
Still some mini improvment

File: src/EventTrackerService.php [hunk 6]
```diff
@@ -40,19 +44,51 @@ class EventTrackerService {
    */
   private $eventStorage;
 
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
```
Comment (wanze) [inline src/EventTrackerService.php]: I think the new dependencies should be `private` to be consistent with the other members.

```suggestion
  private $currentUser;
```

**Review by wanze — CHANGES_REQUESTED** [review]
Thanks for implementing this important update! 👍 🎉Just one minor comment regarding the visibility of the introduced dependencies, then let's merge :)

**Pull request merged**